src/blueapi/service/interface.py

```diff
@@ -76,6 +80,35 @@ def stomp_client() -> StompClient | None:
         return None
 
 
+@cache
+def numtracker_client() -> NumtrackerClient | None:
+    conf = config()
+    if conf.numtracker is not None:
+        if conf.env.metadata is not None:
+            client = NumtrackerClient(url=conf.numtracker.url, headers={})
+            return client
+        else:
+            raise InvalidConfigError(
+                "Numtracker url has been configured, but there is no instrument or"
+                " instrument_session in the environment metadata"
+            )
```

Comment:
This smells a little to me- if the numtracker also needs these fields to be set, shouldn't they be in the numtracker config?

Reply:
Potentially, do we think we might use this config for things that aren't numtracker in the future?

Reply:
I thought the same thing. While this is coupled should we make it so? There was some vision of re-configuring visit within nested plans which is why i kept them apart initially but we could always un-pick it later with that subsequent ticket?

Reply:
If we couple them now and decouple them later it would be a breaking API change, which is nice to avoid. I think leave this as-is but if @DiamondJoseph wants to defend his point further we can do that.

Reply:
Going to leave it to avoid an API change later .